ex4B/ex4.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Write a function that convers a string into number; return null if the string
+ * doesn't contain a valid number.
+ *
+ * For example:
+ * toNumber("71") === 71
+ * toNumber("a7") === null
+ */
+function toNumber(str) {
+
+	var tmpStr = '', i = 0, dotRepeat = null;
```

Comment:
if `dotRepeats` acts as a boolean (it should), initialize it to `false` rather than `null` (even though they work the same).